Table of Contents plugin

carl-alberto opened the pull request:
Added the initial version of the TOC plugin, this is mostly based from the [Handbook ](http://meta.svn.wordpress.org/sites/trunk/wordpress.org/public_html/wp-content/plugins/handbook/) code but slightly modified it.

As I have checked, the heading tags that we have in helphub is using "span id", that is the one that I used for the anchor tags, all we need to finalize is where should we properly place it and the styles involved

cralberto11 committed:
Add starter plugin basic files

cralberto11 committed:
Merge branch 'master' into feature-toc

cralberto11 committed:
Added the TOC code in the start of WP content

cralberto11 committed:
a quick fix for the fatal error, initial version, code still needs cleanup

cralberto11 committed:
Merge branch 'master' into feature-toc

cralberto11 committed:
- Structured the TOC plugin inside a class
- Added the top function inside the the_content filter

Kenshino reviewed (changes requested):
@carl-alberto : Fix the errors brought up by Travis?

cralberto11 committed:
Tried to eliminate travis errors

cralberto11 committed:
Fixed 20
| ERROR | [x] Inline control structures are not allowed
    |       |     (Generic.ControlStructures.InlineControlStructure.NotAllowed)
 31 | ERROR | [ ] Function name "Table_Of_Contents_Lite" is not in snake
    |       |     case format, try "table_of_contents_lite"
    |       |     (WordPress.NamingConventions.ValidFunctionName.FunctionNameInvalid)
 31 | ERROR | [x] Space between function name and opening parenthesis is
    |       |     prohibited.
    |       |     (WordPress.WhiteSpace.ControlStructureSpacing.SpaceBeforeFunctionOpenParenthesis)

cralberto11 committed:
Tried to elinminate
 2 | ERROR | [ ] Missing short description in doc comment
   |       |     (Generic.Commenting.DocComment.MissingShort)
 9 | ERROR | [x] Additional blank lines found at end of doc comment
   |       |     (Generic.Commenting.DocComment.SpacingAfter)

cralberto11 committed:
tried fixing
3 | ERROR | [x] File must end with a newline character
   |       |     (Generic.Files.EndFileNewline.NotFound)
 3 | ERROR | [ ] You must use "/**" style comments for a file comment
   |       |     (Squiz.Commenting.FileComment.WrongStyle)
 3 | ERROR | [ ] Inline comments must end in full-stops, exclamation
   |       |     marks, or question marks
   |       |     (Squiz.Commenting.InlineComment.InvalidEndChar)
 3 | ERROR | [x] There must be no blank line following an inline
   |       |     comment (Squiz.Commenting.InlineComment.SpacingAfter)

cralberto11 committed:
tried to fix the FOUND 81 ERRORS AFFECTING 54 LINES

cralberto11 committed:
cleared up unneeded files

cralberto11 committed:
fixed formatting errors in /plugins/table-of-contents-lite/table-of-contents-lite.php

cralberto11 committed:
fixed php formatting issues in table-of-contents-lite/includes/class-table-of-contents-lite.php

cralberto11 committed:
Fixed travis issues in .table-of-contents-lite/includes/class-table-of-contents-lite.php

cralberto11 committed:
fixed additional issues in table-of-contents-lite/includes/class-table-of-contents-lite.php

cralberto11 committed:
fixing issues table-of-contents-lite/includes/class-table-of-contents-lite.php

cralberto11 committed:
Fixed all Travis CI errors

Kenshino reviewed (changes requested):
Fix the author details in your Pot file

Clorith reviewed (changes requested):
The plugin textdomain should not be loaded before the `init` hook.

When creating the ToC we should escape the title, both the attribute link, and the name of it, as we cannot guarantee something else hasn't filtered the content.

It would be nice if we use `sprintf` functions when combining strings with variables/function results for readability (this is done in some paces, but not all).

cralberto11 committed:
Merge branch 'master' into feature-toc

cralberto11 committed:
Updated pot details and plugin description

cralberto11 committed:
the
	$this->load_plugin_textdomain();
is now loaded after the ‘init’ hook below
	add_action( 'init', array( $this, 'load_localisation' ), 0 );

cralberto11 committed:
utilized sprint every time there are needed outputs

cralberto11 committed:
further cleaned up the pot file

cralberto11 committed:
- Improved initialization procedure
- cleaned up the loading of text domain
